src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingAssessmentResource.java

```diff
@@ -197,6 +204,17 @@ public ResponseEntity<Result> saveProgrammingAssessment(@PathVariable Long parti
                 || newManualResult.getParticipation().getExercise().getAssessmentDueDate().isBefore(ZonedDateTime.now()))) {
             messagingService.broadcastNewResult(newManualResult.getParticipation(), newManualResult);
         }
+
+        var isManualFeedbackRequest = programmingExercise.getAllowManualFeedbackRequests() && Objects.nonNull(participation.getIndividualDueDate())
+                && participation.getIndividualDueDate().isBefore(ZonedDateTime.now());
+        var isBeforeDueDate = Objects.nonNull(programmingExercise.getDueDate()) && programmingExercise.getDueDate().isBefore(ZonedDateTime.now());
```

Comment:
Does that mean the repo will not be unlocked if there is no due date set? Shouldn't it be unlocked in that case as well?

Reply:
imo this whole thing is out of scope, as this PR was supposed to add two buttons.

You cannot set the `allowManualFeedbackRequests` flag without setting an exam due date

Reply:
If the due date should not be null anyway, then I guess this version is okay.